src/gluonts/nursery/duru/data.py

```diff
@@ -0,0 +1,488 @@
+            ),
+            start_time=dataset_val[0]["start"],
+        )  # only works for electricity, because every item has the same start time
+
+        # print("hello")
+
+        # TODO CORRECT?????????????????
+        # dataset_train, dataset_val = gluonts.dataset.DatasetCollection(datasets=dataset_train), gluonts.dataset.DatasetCollection(datasets=dataset_val)
+
+        # doing a rolling prediction dataset for validation data
+        # TODO similarly do for test data
+        # TODO doesn't work yet
+
+        # entries = list(dataset_original.train)
+        # X_list = [entry['target'] for entry in entries]  # list of time series, each of differnt length (hence, cannot convert to numpy array)
+        # # insert "measurements channel if not available
+        # if X_list[0].ndim == 1:   # only one measurement, otherwise expected that already in (measurements/channel, steps) format
+        #     X_list = [np.expand_dims(x, axis=0) for x in X_list]  # new format: (measurements/channel, steps)
+        # H.n_meas = X_list[0].shape[0]
+        #
+        # train_frac, val_frac, test_frac = .7, .15, .15
+        # assert train_frac+val_frac+test_frac == 1.
+        # # train-val-test split s.t. the sub-datasets are non-overlapping in terms of the original time series
+        # X_train = X_list[:int(train_frac*len(X_list))]
+        # X_val = X_list[int(train_frac*len(X_list)):int((train_frac+val_frac)*len(X_list))]
+        # X_test = X_list[int((train_frac+val_frac)*len(X_list)):int((train_frac+val_frac+test_frac)*len(X_list))]
+
+    elif H.dataset == "sine":
+
+        # OLD -----
+        # class SineWaveGluonTSDataset(gluonts.dataset.Dataset):
+        #     def __init__(self):
+        #         self.n_items = 5000
+        #         x = np.linspace(0, 20 * np.pi, 1000)
+        #         sin_x = np.sin(x)
+        #         # add noise
+        #         sin_x = sin_x + np.random.normal(loc=0., scale=0.3, size=sin_x.shape)
+        #         # convert to float32, s.t. later compatible with torch model which expects 32 precision
+        #         sin_x = sin_x.astype(np.float32)
+        #
+        #         self.data = []
+        #         # all items are ientical
+        #         for t in range(self.n_items):
+        #             item = {}
+        #             item['start'] = 0.  # dummy data
+        #             item['target'] = sin_x
+        #             item['item_id'] = str(t)
+        #             self.data.append(item)
+        #
+        #     def __iter__(self):
+        #         for t in range(self.n_items):
+        #             # print(t)
+        #             yield self.data[t]
+        #         # raise StopIteration
+        #
+        #     def __len__(self):
+        #         return self.n_items
+        # OLD -----
+
+        # NEW 1: different noise, different starting point -----
+        class SineWaveGluonTSDataset(gluonts.dataset.Dataset):
```

Comment:
I think we had issues with deriving from `Dataset` in the past. The idea is that it's a protocol (i.e. nominal typing), meaning that you don't need to derive from it.

Reply:
it works right now, so I'd rather like to leave it. It's purpose is to overwrite and expect the __iter__ and __len__ method.